lib/phoenix_live_view/html_algebra.ex

```diff
@@ -108,7 +108,7 @@ defmodule Phoenix.LiveView.HTMLAlgebra do
             ""
         end
 
-      text_starts_with_line_break?(next_node) and text_ends_with_line_break?(next_node) ->
+      tag_block?(prev_node) and !tag_block?(next_node) ->
```

Comment:
To be fair, not sure if that's the right fix. Here is what is happening:

* first pass the strong tag is considered `inline` because it doesn't have line breaks. Therefore it will put the `baz` word right next to it because we use `flex_block` and it still fits the line.
* When running the formatter again, the strong tag is considered `block` because it now has line breaks. So the `baz` word will be pushed to next line.

Another way of fixing that would trying to predict if a `tag_block` will get line breaks if that's possible somehow 🤔

Reply:
> Another way of fixing that would trying to predict if a tag_block will get line breaks if that's possible somehow 🤔

It depends. Sometimes we know it will get line breaks because we force it. But if it depends on the size of the text, then that's impossible to know.